master/internal/db/postgres_trial.go

```diff
@@ -23,53 +23,69 @@ import (
 // TODO: rename raw_steps.
 
 // AddTrial adds the trial to the database and sets its ID.
-func (db *PgDB) AddTrial(trial *model.Trial) error {
+func AddTrial(ctx context.Context, trial *model.Trial, taskID model.TaskID) error {
 	if trial.ID != 0 {
 		return errors.Errorf("error adding a trial with non-zero id %v", trial.ID)
 	}
 
-	if err := db.namedGet(&trial.ID, `
-INSERT INTO trials
-(task_id, request_id, experiment_id, state, start_time, end_time,
-hparams, warm_start_checkpoint_id, seed)
-VALUES (:task_id, :request_id, :experiment_id, :state, :start_time,
-	:end_time, :hparams, :warm_start_checkpoint_id, :seed)
-RETURNING id`, trial); err != nil {
-		// Assume the foreign key constraint is handled by the database.
-		return errors.Wrapf(err, "error inserting trial %v", *trial)
+	err := Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
+		if _, err := tx.NewInsert().Model(trial).Returning("id").Exec(ctx); err != nil {
+			return fmt.Errorf("inserting trial model: %w", err)
+		}
+
+		// Since we are inserting trial TaskRunID == 0 since this is the first run.
+		trialTaskID := &model.TrialTaskID{TrialID: trial.ID, TaskID: taskID, TaskRunID: 0}
```

Comment:
Yeah it is for ordering, I wanted to avoid a lot of the baggage with task / allocation start and end times in our system being unreliable.

Though I see task's start time is not null and seems obviously correct. Switched to ordering based on task start time.